5-auth/news-aggregator/src/helpers/cacheHelper.js

```diff
@@ -0,0 +1,46 @@
+const Redis = require("ioredis");
+const redis = new Redis();
+
+// Add an error event listener
+redis.on("error", (err) => {
+  //console.error("Redis connection error:", err);
+});
+
+const getWithTimeout = (key, ms) => {
+  return new Promise((resolve, reject) => {
+    redis.get(key, (err, result) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(result);
+      }
+    });
+
+    setTimeout(() => {
+      reject(new Error(`Timeout exceeded: ${ms} ms`));
+    }, ms);
+  });
+};
+
+const getNewsFromCache = async (key) => {
+  try {
+    const data = await getWithTimeout(key, 500);
+    return JSON.parse(data);
+  } catch (err) {
+    //console.error("Error retrieving news from cache:", err);
+    return null;
+  }
+};
+
+const setNewsToCache = async (key, data) => {
+  try {
+    await redis.set(key, JSON.stringify(data), "EX", 60 * 60);
+  } catch (err) {
+    //console.error("Error setting news to cache:", err);
```

Comment:
Instead of only catching exception, we can log it, emit metrics (right now, we haven't learnt that in course, but we can look into it for personal experience. You can look in datadog